src/internal/restgateway/restgateway.go

```diff
@@ -0,0 +1,114 @@
+package restgateway
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+
+	pfsgw "github.com/pachyderm/pachyderm/v2/src/pfs"
+	ppsgw "github.com/pachyderm/pachyderm/v2/src/pps"
+
+	admingw "github.com/pachyderm/pachyderm/v2/src/admin"
+	authgw "github.com/pachyderm/pachyderm/v2/src/auth"
+	debuggw "github.com/pachyderm/pachyderm/v2/src/debug"
+	enterprisegw "github.com/pachyderm/pachyderm/v2/src/enterprise"
+	identitygw "github.com/pachyderm/pachyderm/v2/src/identity"
+	licensegw "github.com/pachyderm/pachyderm/v2/src/license"
+	proxygw "github.com/pachyderm/pachyderm/v2/src/proxy"
+	transactiongw "github.com/pachyderm/pachyderm/v2/src/transaction"
+	versiongw "github.com/pachyderm/pachyderm/v2/src/version/versionpb"
+	workergw "github.com/pachyderm/pachyderm/v2/src/worker"
+)
+
+func NewRestGatewayMux(pachClientFactory func(context.Context) *client.APIClient) http.Handler {
```

Comment:
Also, since this is in package `restgateway`, the method can just be named `NewMux`.  https://github.com/golang/go/wiki/CodeReviewComments#package-names

Reply:
done